core/iwasm/common/wasm_runtime_common.c

```diff
@@ -248,6 +249,15 @@ runtime_signal_handler(void *sig_addr)
 
         if (is_sig_addr_in_guard_pages(sig_addr, module_inst)) {
             wasm_set_exception(module_inst, "out of bounds memory access");
+#if WASM_ENABLE_MULTI_MODULE != 0
+            if (jmpbuf_node->module_inst
+                && jmpbuf_node->module_inst
+                       != (WASMModuleInstanceCommon *)module_inst) {
+                wasm_runtime_propagate_exception_from_import(
+                    (WASMModuleInstanceCommon *)jmpbuf_node->module_inst,
+                    (WASMModuleInstanceCommon *)module_inst);
+            }
+#endif
```

Comment:
i'm not sure why we need WASM_ENABLE_MULTI_MODULE handling in this low level logic.

exception propagation should be done regardless of hw bound check, shouldn't it?

Reply:
This specific case #4703 occurs only when an exception happens in the child module instance, and the main module instance fails to copy that exception, causing the problem.

We both agree that there is [another simple solution](https://github.com/bytecodealliance/wasm-micro-runtime/pull/4714#pullrequestreview-3533295845) for this issue.

However, in the short term, I can't determine under which circumstances it is necessary to switch module instances before calling a WASM function. Please enlighten me.